src/main/java/io/github/jhipster/online/security/jwt/TokenProvider.java

```diff
@@ -104,8 +104,8 @@ public String createToken(Authentication authentication, boolean rememberMe) {
     }
 
     public Authentication getAuthentication(String token) {
-        Claims claims = Jwts.parser()
-            .setSigningKey(key)
+        Claims claims = Jwts.parserBuilder()
+            .setSigningKey(key).build()
```

Comment:
If I read correctly their doc and code, `Jwts.parserBuilder().setSigningKey(key).build()` is thread-safe and so can be shared and can be final rather than called each time.

https://github.com/jwtk/jjwt/blob/master/README.md#reading-a-jws

https://github.com/jwtk/jjwt/blob/master/impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtParserBuilder.java#L199

https://github.com/jwtk/jjwt/issues/473

Reply:
Although I haven't tested it

Reply:
@gmarziou : The problem here is that we initialize the key post construct. And I am not sure how we can make this final and static unless we could make the key also final. Do you have any idea? 🤔

Reply:
Well I don't understand why there is a `@PostConstruct init()`, to me all code in init() should be in the constructor and it should build the parser.

Reply:
@gmarziou : I've refactored the class to remove the @PostConstruct and put all code in the constructor; and it seems to work fine. So we don't need a @PostConstruct here as we don't do any dependency injection; that's the idea right; am I correct?

Reply:
@gmarziou : Although the application works fine I think this was done that way because of the tests maybe. The tests seems to fail as when the constructor is invoked the values are null. Is there a way where we could set the values while initializing the object in junit? I tried some ways but I couldn't still figure out that part. 🤔

Reply:
@gmarziou : Disregard my previous comment; we could make a slight change to the way in which the JUnit tests are arranged to make it work. I think now it works fine. Please let me know if you see any other changes to be made. 😄